M/MAiNGO/build_tarballs.jl

```diff
@@ -0,0 +1,85 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+name = "MAiNGO"
+version = v"0.7.2"
+
+# Collection of sources required to complete build
+sources = [
+    GitSource("https://git.rwth-aachen.de/avt-svt/public/maingo.git", "252733413a29dbe5b84a4cdaf53e60e9934f372f"),
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir/maingo/
+git remote set-url origin https://git.rwth-aachen.de/avt-svt/public/maingo.git
+mkdir build
+cd build
+git submodule init
+git submodule update -j 1
+
+
+common_cmake_options="-DCMAKE_BUILD_TYPE=Release \
+                     -DMAiNGO_build_standalone=True \
+                     -DMAiNGO_build_shared_c_api=True \
+                     -DMAiNGO_build_parser=True \
+                     -DMAiNGO_use_cplex=False \
+                     -DMAiNGO_use_melon=False"
+
+# GCC used because of https://github.com/JuliaPackaging/Yggdrasil/issues/7139
+if [[ "${target}" == x86_64-apple-darwin* ]]; then
+    export MACOSX_DEPLOYMENT_TARGET=10.15
+    cmake -DCMAKE_INSTALL_PREFIX=${prefix} \
+          -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN%.*}_gcc.cmake \
```

Comment:
Why not clang?  Also, at this point you could avoid the conditional and use gcc everywhere, but I'd prefer to understand what's wrong with clang, that's usually the best choice for macOS.